lib/parse.js

```diff
@@ -1051,32 +1124,34 @@ function separatorChar (c) {
     console.warn(`JSON5: '${formatChar(c)}' in strings is not valid ECMAScript; consider escaping`)
 }
 
-function formatChar (c) {
+function formatChar (n) {
+    const c = typeof n === 'string' ? n.codePointAt(0) : n
+
     const replacements = {
-        "'": "\\'",
-        '"': '\\"',
-        '\\': '\\\\',
-        '\b': '\\b',
-        '\f': '\\f',
-        '\n': '\\n',
-        '\r': '\\r',
-        '\t': '\\t',
-        '\v': '\\v',
-        '\0': '\\0',
-        '\u2028': '\\u2028',
-        '\u2029': '\\u2029',
+        0x0027 /* "'" */: "\\'",
```

Comment:
```suggestion
        0x0027 /* "'" */: String.raw`\'`,
```
Easier to read. 
Hopefully resolved at JS engine parsing time (I hope you know which engine and with parser I mean :-) and does not ruin performance again.

Reply:
I'll leave that part as I found it.